generators/python/core_utilities/shared/file.py

```diff
@@ -39,3 +39,22 @@ def convert_file_dict_to_httpx_tuples(
         else:
             httpx_tuples.append((key, file_like))
     return httpx_tuples
+
+
+def with_content_type(*, file: File, content_type: str) -> File:
+    """ """
+    if isinstance(file, tuple):
+        if len(file) == 2:
+            filename, content = cast(Tuple[Optional[str], FileContent], file)
+            return (filename, content, content_type)
+        elif len(file) == 3:
```

Comment:
It feels a bit odd that the input to these could be:
```
with_content_type(file=("file.json", <bytes>, "application/json"), content_type="application/json")
```

Since the tuple already allows for overriding the content type


It might make more sense to be something like

```
def construct_file(
    *,
    file_contents: FileContent,
    file_name: Optional[str]=None,    
    content_type: Optional[str]=None,
    headers: Optional[Mapping[str,str]]=None
```
So that you're effectively just passing in a well typed and documented `core.File` constructor

Reply:
Unless this is strictly internal

Reply:
Yeah its a core utility thats strictly internal

Reply:
is there a safe way to index into tuples to call into that if i changed the method signature?

Reply:
going to merge to get out of incident, but happy to break this since its internal